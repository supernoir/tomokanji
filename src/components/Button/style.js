import styled from 'styled-components';
import { Button } from './index';

export const StyledButton = styled(Button)`
  background: ${props => props.theme.white};
  color: ${props => props.theme.primary};
  min-width: 120px;
  font-weight: 400;
  font-size: 1.2rem;
  padding: .4em .8em;
  margin: .8em 0.6em;
  border-radius: .2em;
  cursor: pointer;
  text-decoration: none;
  transition: background 0.4s ease;
  box-shadow: 0px 4px 2px ${props => props.theme.primary};

  &:hover {
    color: ${props => props.theme.secondary};
    box-shadow: 0px 4px 2px ${props => props.theme.secondary};
    transition: box-shadow 0.4s ease-in-out;
  }
`;

export const StyledRandomButton = styled(Button)`

  background: ${props => `linear-gradient(135deg, ${props.theme.secondary}, ${props.theme.primary})`};
  color: ${props => props.theme.white};
  border: 3px solid ${props => props.theme.white};
  min-width: 120px;
  font-weight: 400;
  font-size: 1.2rem;
  padding: .4em .8em;
  margin: .8em 0.6em;
  border-radius: .2em;
  cursor: pointer;
  text-decoration: none;
  transition: background 0.4s ease;
  box-shadow: 0px 4px 2px ${props => props.theme.primary};


    &:hover {
      background: ${props => `linear-gradient(-135deg, ${props.theme.secondary}, ${props.theme.primary})`};
      box-shadow: 0px 4px 2px ${props => props.theme.secondary};
      transition:  box-shadow 0.4s ease-in-out;
      animation-name: shake;
      animation-duration: 80ms;
      animation-iteration-count: infinite;
      animation-direction: alternate
    }
    @keyframes shake {
      from {
        transform: rotate(-2deg);
      }
      to {
        transform: rotate(2deg);
      }
      
    }
`;
