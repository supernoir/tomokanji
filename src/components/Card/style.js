import styled from 'styled-components';

export const StyledCard = styled.div`
  background: ${props => props.theme.black};
  background: rgba(0,0,0, 0.4);
  color: ${props => props.theme.black};
  padding: 1em;
  border-radius: .4em;
`;

export const StyledCardClassification = styled.p`
  margin-left: 0.6em;
  border-radius: 0.4em;
  padding: 0.2em 0.4em;
  background: ${props => props.theme.white};
  color: ${props => props.theme.black};
`;

export const StyledCardKanjiCount = styled.p`
  background: pink;
`;

export const StyledCardClassAndCount = styled.div`
  border-radius: 4px;
  border: 1px solid ${props => props.theme.white};
  margin-top: 0.8em;
  min-width: 4em;
  padding: 0.2em 0.8em;
  color: ${props => props.theme.white};
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const StyledCardDetails = styled.div`
  width: 100%;
  font-size: 1.2em;
  font-weight: 300em;
  color: ${props => props.theme.tertiary};
`;

export const StyledCardMeaning = styled.div`
  width: 100%;
  font-size: 1.8em;
  font-weight: 500;
`;

export const StyledCardFooter = styled.div`
  background: rgba(0,0,0,0.2);
  color: ${props => props.theme.white};
  padding: 2em 0.4em;
  border-radius: .4em;
  display: flex;
  flex-flow: column;
  align-items: center;
  `;

export const StyledCardKanjiContent = styled.h1`
  color: ${props => props.theme.black};
  font-size: 12em;
  line-height: 0em;
  font-weight: 300;
`;

export const StyledCardKanji = styled.div`
  background: ${props => props.theme.white};
  border-radius: .4em;
  text-align: center;
  margin: 0;
  padding: 2em;
`;

export const StyledCardBody = styled.div`
  margin: 1em 0;
  padding: 0;
`;

export const StyledCardOnyomiReadingContent = styled.p`
  font-size: 2em;
  line-height: 1.2em;
  text-shadow: 0 1px 0 ${props => props.theme.onyomi};
  background: rgba(220, 120, 80, 0.4);
  border-radius: 1em;
`;
export const StyledCardKunyomiReadingContent = styled.p`
  font-size: 2em;
  line-height: 1.2em;
  text-shadow: 0 1px 0 ${props => props.theme.kunyomi};
  background: rgba(80, 180, 160, 0.4);
  border-radius: 1em;
`;

export const StyledCardOnyomiReadingType = styled.p`
  color: ${props => props.theme.white};
  font-size: 0.8em;
  line-height: 0em;
  text-shadow: 0 1px 2px ${props => props.theme.darkonyomi};
`;

export const StyledCardKunyomiReadingType = styled.p`
  color: ${props => props.theme.white};
  font-size: 0.8em;
  line-height: 0em;
  text-shadow: 0 1px 2px ${props => props.theme.darkkunyomi};
`;

export const StyledCardOnyomi = styled.div`
  background: ${props => `linear-gradient(-30deg, ${props.theme.onyomi}, ${props.theme.lightonyomi})`};
  color: ${props => props.theme.white};
  padding: 1em;
  margin-left: 0.5em;
  width: 50%;
  height: 100%;
  border-radius: 0.2em;
  border-top: 6px solid ${props => props.theme.onyomi};
`;

export const StyledCardKunyomi = styled.div`
  background: ${props => `linear-gradient(-30deg, ${props.theme.kunyomi}, ${props.theme.lightkunyomi})`};
  color: ${props => props.theme.white};
  padding: 1em;
  margin-left: 0.5em;
  width: 50%;
  height: 100%;
  border-radius: 0.2em;
  border-top: 6px solid ${props => props.theme.kunyomi};
`;

export const StyledCardHead = styled.div`
  display: flex;
  align-content: space-between;
  margin: 0;
  height: 100%;
`;
