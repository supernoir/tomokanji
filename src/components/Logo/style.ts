import styled from 'styled-components';
import { Logo, Highlight, Brand } from './index';

export const StyledHeader = styled.header`
  grid-row: 1;
  grid-column: 1;
  background: ${props => props.theme.black};
  opacity: 0.8;
  height: 3.5em;
  display: flex;
  justify-content: center;
`;

export const StyledBrand = styled(Brand)`
  display: flex;
  flex-flow: row;
  padding: 0.2em;
`;

export const StyledLogo = styled(Logo)`
    margin: 0.2em 0.4em;
    color: ${props => props.theme.white};
    font-weight: light;
    font-size: 18px;
    text-shadow: 0px 2px 3px ${props => props.theme.black};
`;

export const StyledHighlight = styled(Highlight)`
  color: ${props => props.theme.tertiary};
  font-weight: bold;
`;
