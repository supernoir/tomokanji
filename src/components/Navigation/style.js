import styled from 'styled-components';
import { Navigation } from './index';

export const StyledNavigation = styled(Navigation)`
  margin: 2em 0 4em;
  display: flex;

  @media screen and (min-width: 480px) {
    flex-direction: row;
  }

  @media screen and (max-width: 480px){
    flex-direction: column;
  }
`;
