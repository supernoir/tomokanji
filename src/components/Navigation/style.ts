import styled from 'styled-components';

export const StyledNavigation = styled.aside`
  margin: 2em 0 4em;
  display: flex;
  width: 100%;
  justify-content: center;
  
  @media screen and (min-width: 480px) {
    flex-direction: row;
  }

  @media screen and (max-width: 480px){
    flex-direction: column;
  }
`;
