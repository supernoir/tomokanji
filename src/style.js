import styled from 'styled-components';
import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
html, body {
  @import url('https://fonts.googleapis.com/css?family=Noto+Sans');
  font-family: "Noto Sans", sans-serif;
  background: ${props => props.theme.primary};
  margin: 0
  padding: 0
  zoom: 1;
  color: ${props => props.theme.white};
  text-align: center;
} 
`;

export const StyledApp = styled.div`
  color: ${props => props.theme.white};
  text-align: center;
  height: 100%;
  display: grid;
  grid-template-rows: 1,2;
  grid-template-columns: 1;
  background: ${props => `linear-gradient(15deg, ${props.theme.primary}, ${props.theme.secondary})`};
`;

export const StyledPage = styled.main`
  grid-row: 2;
  grid-column: 1;
  margin: 10%;
  text-align: center;
  border-radius: .4em;
`;

createGlobalStyle`${GlobalStyle}`;
