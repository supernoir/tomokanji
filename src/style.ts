import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle: any = createGlobalStyle`
@import url('https://fonts.googleapis.com/css?family=Noto+Sans');
body {
  font-family: "Noto Sans", sans-serif;
  background: rgb(180, 80, 120);
  margin: 0;
  padding: 0;
  zoom: 1;
  color: white;
  color: white;
  text-align: center;
} 
`;

export const StyledApp = styled.div`
  color: ${props => props.theme.white};
  text-align: center;
  height: 100%;
  display: grid;
  grid-template-rows: 1,2;
  grid-template-columns: 1;
  background: ${props => `linear-gradient(15deg, ${props.theme.primary}, ${props.theme.secondary})`};
`;

export const StyledPage = styled.main`
  grid-row: 2;
  grid-column: 1;
  margin: 10%;
  text-align: center;
  border-radius: .4em;
`;