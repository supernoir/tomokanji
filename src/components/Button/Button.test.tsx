import React from 'react';
import { StyledButton, StyledRandomButton } from './style';
import Renderer from 'react-test-renderer';

describe('Button', () => {
	const testClickHandler = jest.fn();
	test('should match the snapshot', () => {
		const component = Renderer.create(<StyledButton label={'example'} handleClick={testClickHandler} />);
		expect(component).toMatchSnapshot();
	});
});

describe('RandomButton', () => {
	const testClickHandler = jest.fn();
	test('should match the snapshot', () => {
		const component = Renderer.create(<StyledRandomButton label={'example'} handleClick={testClickHandler} />);
		expect(component).toMatchSnapshot();
	});
});