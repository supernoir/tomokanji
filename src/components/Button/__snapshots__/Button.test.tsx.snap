// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<a
  className="sc-bdVaJa hPSpy"
  onClick={[MockFunction]}
>
  example
</a>
`;

exports[`RandomButton should match the snapshot 1`] = `
<a
  className="sc-bwzfXH hufrb"
  onClick={[MockFunction]}
>
  example
</a>
`;
